File: src/main/java/jb/tester/IntegratedScoreTester.java
package jb.tester;

import jb.config.Opts;
import jb.data.Dataset;
import jb.data.IntegratedModel;
import jb.data.Problem;
import jb.data.ValidatingTestingTuple;

import static jb.util.ModelUtils.getIndexOfSubspace;

public class IntegratedScoreTester {

    public double test(IntegratedModel integratedModel, ValidatingTestingTuple validatingTestingTuple, Opts opts) {
        double counter = 0;
        Problem problem = validatingTestingTuple.getTestingProblem();
        for (int i = 0; i < problem.getY().length; i++) {
            int index = getIndexOfSubspace(opts.getNumberOfSpaceParts(), problem.getX()[i][0], validatingTestingTuple.getMinX(), validatingTestingTuple.getMaxX());
            double value = integratedModel.getA()[index] * problem.getX()[i][0] + integratedModel.getB()[index];
            if (value > problem.getX()[i][1] ^ problem.getY()[i] == 1) {
                counter++;
            }
        }
        double score = counter / problem.getY().length;
        score = score > .5 ? score : 1 - score;
        return score;
    }
}

File: src/main/kotlin/jb/tester/IntegratedScoreTester.kt
package jb.tester

import jb.config.Opts
import jb.data.IntegratedModel
import jb.data.Problem
import jb.data.ValidatingTestingTuple

import jb.util.ModelUtils.getIndexOfSubspace

class IntegratedScoreTester {

    fun test(integratedModel: IntegratedModel, validatingTestingTuple: ValidatingTestingTuple): Double {
        var counter = 0.0
        val (x, y) = validatingTestingTuple.testingProblem
        for (i in 0 until y.size) {
            val index = getIndexOfSubspace(integratedModel.x, x[i][0])
            val value = integratedModel.a[index] * x[i][0] + integratedModel.b[index]
            if ((value > x[i][1]) xor (y[i] == 1)) {
                counter++
            }
        }
        var score = counter / y.size
        score = if (score > .5) score else 1 - score
        return score
    }
}
